# Eksplorasi Dataset Wine

Notebook ini digunakan untuk memahami struktur Dataset Wine sebelum melakukan eksperimen Machine Learning.

Tujuan notebook:
- Memuat dataset Wine
- Menentukan nama setiap kolom
- Melihat struktur dan ukuran dataset
- Memisahkan fitur (`X`) dan label (`y`)
- Melihat distribusi masing-masing kelas

## 1. Memuat Dataset

Dataset Wine disimpan dalam file `wine.data`.

File ini tidak memiliki nama kolom pada baris pertama, sehingga nama kolom perlu ditentukan secara manual berdasarkan dokumentasi dataset.

In [20]:
import pandas as pd

columns = [
    "class",
    "alcohol",
    "malic_acid",
    "ash",
    "alcalinity_of_ash",
    "magnesium",
    "total_phenols",
    "flavanoids",
    "nonflavanoid_phenols",
    "proanthocyanins",
    "color_intensity",
    "hue",
    "od280_od315",
    "proline"
]

df = pd.read_csv("../../data/wine/wine.data", header=None, names=columns) 

## 2. Melihat Data Awal

Lima baris pertama ditampilkan untuk memastikan dataset berhasil dibaca dan struktur kolom sudah sesuai.

In [21]:
df.head()

,class,alcohol,malic_acid,ash,alcalinity_of_ash,magnesium,total_phenols,flavanoids,nonflavanoid_phenols,proanthocyanins,color_intensity,hue,od280_od315,proline
0,1,14.23,1.71,2.43,15.6,127,2.80,3.06,0.28,2.29,5.64,1.04,3.92,1065
1,1,13.20,1.78,2.14,11.2,100,2.65,2.76,0.26,1.28,4.38,1.05,3.40,1050
2,1,13.16,2.36,2.67,18.6,101,2.80,3.24,0.30,2.81,5.68,1.03,3.17,1185
3,1,14.37,1.95,2.50,16.8,113,3.85,3.49,0.24,2.18,7.80,0.86,3.45,1480
4,1,13.24,2.59,2.87,21.0,118,2.80,2.69,0.39,1.82,4.32,1.04,2.93,735


## 3. Melihat Ukuran Dataset

Ukuran dataset diperiksa untuk mengetahui jumlah sampel dan jumlah kolom yang tersedia.

In [22]:
df.shape

(178, 14)

## 4. Memisahkan Fitur dan Label

Dalam Machine Learning, dataset dipisahkan menjadi:

- `X` = fitur atau informasi yang digunakan model untuk belajar.
- `y` = label atau jawaban benar yang ingin diprediksi oleh model.

Pada Dataset Wine, kolom `class` merupakan label, sedangkan 13 kolom lainnya merupakan fitur.

In [23]:
X = df.drop("class", axis = 1)
y = df["class"]

### Contoh fitur

In [24]:
X.head()

,alcohol,malic_acid,ash,alcalinity_of_ash,magnesium,total_phenols,flavanoids,nonflavanoid_phenols,proanthocyanins,color_intensity,hue,od280_od315,proline
0,14.23,1.71,2.43,15.6,127,2.80,3.06,0.28,2.29,5.64,1.04,3.92,1065
1,13.20,1.78,2.14,11.2,100,2.65,2.76,0.26,1.28,4.38,1.05,3.40,1050
2,13.16,2.36,2.67,18.6,101,2.80,3.24,0.30,2.81,5.68,1.03,3.17,1185
3,14.37,1.95,2.50,16.8,113,3.85,3.49,0.24,2.18,7.80,0.86,3.45,1480
4,13.24,2.59,2.87,21.0,118,2.80,2.69,0.39,1.82,4.32,1.04,2.93,735


### Contoh label

In [25]:
y.head()

0    1
1    1
2    1
3    1
4    1
Name: class, dtype: int64

## 5. Distribusi Kelas

Distribusi kelas diperiksa untuk melihat jumlah sampel pada masing-masing kelas Wine.

In [26]:
y.value_counts().sort_index()

class
1    59
2    71
3    48
Name: count, dtype: int64

## 6. Pemeriksaan Data Kosong

Data kosong atau `missing value` perlu diperiksa karena dapat memengaruhi proses training model.

In [27]:
df.isnull().sum()

class                   0
alcohol                 0
malic_acid              0
ash                     0
alcalinity_of_ash       0
magnesium               0
total_phenols           0
flavanoids              0
nonflavanoid_phenols    0
proanthocyanins         0
color_intensity         0
hue                     0
od280_od315             0
proline                 0
dtype: int64

## Kesimpulan Eksplorasi Awal

Dataset Wine memiliki 178 sampel dengan 13 fitur numerik dan 1 label kelas.

Dataset memiliki tiga kelas dan tidak ditemukan nilai kosong pada data.

Selanjutnya dataset dapat digunakan untuk eksperimen klasifikasi menggunakan Decision Tree, K-Nearest Neighbors, dan Random Forest.